.github/workflows/test-deploy.yml

```diff
@@ -0,0 +1,145 @@
+name: Test & Deploy
+on:
+  schedule:
+    - cron: 0 0 * * *
+  release:
+    types: [published]
+  pull_request:
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2
+      - run: npm ci
+      - run: npm run lint
+  test:
+    environment:
+      ${{ (github.event_name == 'pull_request' &&
+      github.event.pull_request.head.repo.full_name != github.repository) &&
+      'test-external' || 'test-internal' }}
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2
+      - uses: actions/setup-python@v2
+      - run: pip install tensorboard
+      - run: npm ci
+      - run: npm run test
+        env:
+          GITHUB_TOKEN: ${{ github.token }}
+          TEST_GITHUB_TOKEN: ${{ secrets.TEST_GITHUB_TOKEN }}
+          TEST_GITHUB_REPO: https://github.com/iterative/cml_qa_tests_dummy
+          TEST_GITHUB_SHA: 62edc8b3f46a60b3fe1e5c08fd3e0046d350ee29
+          TEST_GITLAB_TOKEN: ${{ secrets.TEST_GITLAB_TOKEN }}
+          TEST_GITLAB_REPO: https://gitlab.com/iterative.ai/cml_qa_tests_dummy
+          TEST_GITLAB_SHA: c4c13286e78dc252dd2611f31a755f10d343fbd4
+          TEST_BBCLOUD_TOKEN: ${{ secrets.TEST_BBCLOUD_TOKEN }}
+          TEST_BBCLOUD_REPO: https://bitbucket.org/iterative-ai/cml-qa-tests-dummy
+          TEST_BBCLOUD_SHA: 9bb9131ce0af294fe1c6eedca1f2bce3983e80bd
+  packages:
+    needs: [lint, test]
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2
+        with:
+          registry-url: https://registry.npmjs.org
+      - run: npm install
+      - run:
+          npm ${{ github.event_name == 'release' && 'publish' || 'publish
+          --dry-run' }}
+        env:
+          NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
+  images:
+    runs-on: ubuntu-latest
+    needs: packages
+    strategy:
+      matrix:
+        dvc: [1, 2]
+        base: [0, 1]
+        gpu: [false, true]
+        include:
+          - base: 0
+            ubuntu: 18.04
+            python: 2.7
+            cuda: 10.1
+            cudnn: 7
+          - base: 1
+            ubuntu: 20.04
+            python: 3.8
+            cuda: 11.0.3
+            cudnn: 8
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - name: Metadata
+        id: metadata
+        run: |
+          latest_tag=$(git describe --tags | cut -d- -f1)
+          cml_version=${latest_tag##v}
+          dvc_version=$(python3 -c '
+          from distutils.version import StrictVersion as Ver
+          from urllib.request import urlopen
+          from json import load
+          data = load(urlopen("https://pypi.org/pypi/dvc/json"))
+          print(
+              max(
+                  (i.strip() for i in data["releases"] if i.startswith("${{ matrix.dvc }}")),
+                  default="${{ matrix.dvc }}",
```

Comment:
```suggestion
          ver_pre = "${{ matrix.dvc }}".rstrip(".") + "."
          print(
              max(
                  (i.strip() for i in data["releases"] if i.startswith(ver_pre)),
                  default="${{ matrix.dvc }}",
```

Reply:
@0x2b3bfa0 be happy

Reply:
@casperdcl, now you're going to hate me in every conceivable way:

```python
from packaging.version import Version
from packaging.specifiers import Specifier

major = 1
versions = ["1.0.0", "1.0.1", "2.0.0", "20.0.0"]

def latest(versions, major="*"):
    versions = map(Version, versions)
    return max(filter(Specifier(f"== {major}.*").contains, versions))

for major in 1, 2, 20:
    print(major, latest(versions, major)) 
```

It's true that [`packaging`](https://pypi.org/project/packaging/) is not part of the standard library, but it comes preinstalled in our environment.

I'll spare you the quotations: [1513](https://xkcd.com/1513/), [1833](https://xkcd.com/1833/), [1695](https://xkcd.com/1695/) and [the regrets](https://legacy.python.org/doc/essays/ppt/regrets/PythonRegrets.pdf).

Reply:
btw feel free to open a follow-up PR if you want to mess around with the versions (or anything else) more

Reply:
Definitely not a priority, but might be a good pastime for a rainy weekend. Rest assured that I won't miss the opportunity of using a perceptron to calculate full versions from major versions. 😸